### PR TITLE
feat: Take sampleRate from metadata

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java [hunk 1]
```diff
@@ -44,6 +46,14 @@ static void applyMetadata(Context context, SentryAndroidOptions options) {
         options.getLogger().log(SentryLevel.DEBUG, "isAnrEnabled read: %s", isAnrEnabled);
         options.setAnrEnabled(isAnrEnabled);
 
+        if (options.getSampleRate() == null) {
+          Double sampleRate = metadata.getDouble(SAMPLE_RATE, -1);
+          options.getLogger().log(SentryLevel.DEBUG, "sampleRate read: %s", sampleRate);
+          if (sampleRate != -1) {
+            options.setSampleRate(sampleRate);
+          }
+        }
```
Comment (marandaneto) [inline sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java]: what's about:
`options.setSampleRate(metadata.getDouble(SAMPLE_RATE, options.getSampleRate()));`

Comment (bruno-garcia) [reply sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java]: Throws when getSampleRate is null. I need to know when no value was defined at all so I used `-1` for that

Comment (bruno-garcia) [reply sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java]: Also, I think we need to change the approach on the other one. 

Right now we're overriding any value set on option with the value from metadata. It should be the way I did here where we check first if options doesn't have any settings, and only then read from metadata. 

Separate PR for that though

**Comment by fdrabek** [comment]
Can we please also add release and environment?

File: sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java (hunk 1 above)
Comment (marandaneto) [reply sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java]: > Also, I think we need to change the approach on the other one.
> 
> Right now we're overriding any value set on option with the value from metadata. It should be the way I did here where we check first if options doesn't have any settings, and only then read from metadata.
> 
> Separate PR for that though

options callback is triggered afterward, so no values are replaced unless one bypasses it.

**Pull request merged**